datafusion/sql/tests/cases/plan_to_sql.rs

```diff
@@ -2596,3 +2600,27 @@ fn test_not_ilike_filter_with_escape() {
         @"SELECT person.first_name FROM person WHERE person.first_name NOT ILIKE 'A!_%' ESCAPE '!'"
     );
 }
+
+#[test]
+fn test_struct_expr() {
```

Comment:
Is it possible to test using the non struct syntax too?

Like `SELECT ["ProductName"]["product"]`?